src/gui.ml

```diff
@@ -154,11 +155,51 @@ let draw_position (bd : Game.t) (imgs : image list) =
   let bd = board_to_array bd in
   draw_position_rows bd imgs 0
 
-let draw_game (bd : Game.t) =
+let rec get_potential_squares (move_list : move list) currX currY :
+    (int * int) list =
+  match move_list with
+  | ((a, b), (c, d)) :: t when currX = a && currY = b ->
+      (c, d) :: get_potential_squares t currX currY
+  | h :: t -> get_potential_squares t currX currY
+  | _ -> []
+
+let draw_potential_circles
+    (move_list : move list)
+    (imgs : image list)
+    currX
+    currY =
+  let potential_moves = get_potential_squares move_list currX currY in
+  let rec draw_circle (potential_moves : (int * int) list) =
```

Comment:
I don't think you need to pass in potential_moves here. It is already defined in scope, so you can reference it in the function body. Just pass in a unit.